pkg/controllers/syncer/translator/namespaced_translator.go

```diff
@@ -216,6 +223,58 @@ func translateAnnotations(vObj client.Object, pObj client.Object, excluded []str
 	return retMap
 }
 
+func getActionOrConditionValue(annotation, actionOrCondition string) string {
+	i := strings.Index(annotation, actionOrCondition)
+	if i > -1 {
+		return annotation[i+len(actionOrCondition):]
+	} else {
+		return ""
+	}
+}
+
+func processAlbAnnotations(vObj client.Object, k string, v string) (string, string) {
+	if strings.HasPrefix(k, AlbActionsAnnotation) {
+		type actionPayload struct {
+			Type          string `json:"type"`
+			ForwardConfig struct {
+				TargetGroups                []map[string]interface{} `json:"targetGroups"`
+				TargetGroupStickinessConfig map[string]interface{}   `json:"targetGroupStickinessConfig"`
+			} `json:"forwardConfig"`
+		}
+		// change k
+		action := getActionOrConditionValue(k, ActionsSuffix)
+		k = strings.Replace(k, action, translate.PhysicalName(action, vObj.GetNamespace()), 1)
+		// change v
+		var payload *actionPayload
+		err := json.Unmarshal([]byte(v), &payload)
+		if err != nil {
+			fmt.Println(err)
+		} else {
+			if payload != nil &&
+				&payload.ForwardConfig != nil &&
+				&payload.ForwardConfig.TargetGroups != nil &&
+				len(payload.ForwardConfig.TargetGroups) > 0 {
+				for _, targetGroup := range payload.ForwardConfig.TargetGroups {
+					serviceName := targetGroup["serviceName"].(string)
```

Comment:
Please make a safe cast here as this will panic otherwise if serviceName is not defined or if it is not a string